api/v1beta1/common_validate.go

```diff
@@ -0,0 +1,36 @@
+/*
```

Comment:
Thanks for your review!

I think it's better to have this function here rather than in the "internal" path, because we may need to call it in other places outside of this package. 
For example, here in the package `sigs.k8s.io/cluster-api/test` we also need to use this function for validation:
https://github.com/kubernetes-sigs/cluster-api/blob/f1129eb781ae2b002ae8afa7b0dc4a392ce3537f/test/infrastructure/docker/api/v1beta1/dockerclustertemplate_webhook.go#L66-L67

So, I think it's more appropriate to place this function here for now. What do you think?

Reply:
We should be able to call functions from the top-level internal package across the codebase, so I don't think it would interfere with the `test/infrastructure/docker` usage.

Reply:
Oh, you are right. I didn't realize that earlier. I'll move this function to the internal package.

Reply:
I found another problem.

The function itself requires the package `sigs.k8s.io/cluster-api/api/v1beta1` to be imported, and the webhook interface functions in this package also need to call the `Validate` function. If we put this function in a different package, it will result in a circular dependency between the two packages. 

Therefore, it seems that this function cannot be placed in the internal package.

Reply:
I would suggest to keep the function here for simplicity.

We also have a history of dropping Validate*() functions on API types. So if we have to get rid of it at some point there is precedent for deprecation + removal